packages/docusaurus-theme-classic/src/theme/hooks/useDocs.ts

```diff
@@ -5,27 +5,7 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-// See https://github.com/facebook/docusaurus/issues/3360
-// TODO find a better solution, this shouldn't be needed
-
-// TODO this is not ideal and produce a warning!
-// see https://github.com/webpack/webpack/issues/7713#issuecomment-467888437
-// note: warning can be filtered: https://github.com/facebook/docusaurus/pull/3382#issuecomment-684966924
-try {
-  // eslint-disable-next-line global-require
-  module.exports = require('@theme-init/hooks/useDocs');
-} catch (e) {
-  // In case the docs plugin is not available, might be useful to stub some methods here
-  // https://github.com/facebook/docusaurus/issues/3947
-  const Empty = {};
-  module.exports = {
-    useAllDocsData: () => Empty,
-    useActivePluginAndVersion: () => undefined,
-  };
-}
-
-/*
-throw new Error(
-  "The docs plugin is not used, so you can't require the useDocs hooks. ",
-);
- */
+// Re-expose useDocs
+// Ensure it's always statically available even if user is not using the docs plugin
+// Problem reported for the blog-only mode: https://github.com/facebook/docusaurus/issues/3360
+export * from '@docusaurus/plugin-content-docs/src/theme/hooks/useDocs';
```

Comment:
@SamChou19815 this change can affect the blog-only mode, the e2e test pass and it looks it keeps working, let me know if you see any issue on next release